src/Jobs/RetryFailedJob.php

```diff
@@ -55,10 +56,26 @@ public function handle(Queue $queue, JobRepository $jobs)
      */
     protected function preparePayload($id, $payload)
     {
-        return json_encode(array_merge(json_decode($payload, true), [
+        $payload = json_decode($payload, true);
+
+        return json_encode(array_merge($payload, [
             'id' => $id,
             'attempts' => 0,
             'retry_of' => $this->id,
+            'timeoutAt' => $this->prepareNewTimeout($payload),
         ]));
     }
+
+    /**
+     * Prepare the timeout.
+     *
+     * @param  array  $payload
+     * @return integer|null
+     */
+    protected function prepareNewTimeout($payload)
+    {
+        return $payload['timeoutAt']
+                        ? Chronos::now()->addSeconds(ceil($payload['timeoutAt'] - $payload['pushedAt']))->getTimestamp()
```

Comment:
@themsaid that is a nice solution, very good, thank you !

We had solved this internally by just overriding the `RetryFailedJob` class and setting `timeoutAt` to `null`, but this looks like a better approach indeed. But that's exactly why I haven't pushed a pull request; I didn't feel like I could make the best code decisions while contributing to Horizon just yet.